### PR TITLE
RegisterLookupSummary now is TypeScript

### PR DESCRIPTION
**Pull request by robinpiets**
RegisterLookupSummary now is TypeScript

### CODE REVIEW

**Review by svenjens — APPROVED** [review]

File: apps/client/src/atoms/ComponentWrapper.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { themeSpacing } from "@amsterdam/asc-ui";
 import styled, { css } from "styled-components";
 
 type Props = {
-  marginBottom?: number;
+  marginBottom?: number | undefined;
```
Comment (afjlambert) [inline apps/client/src/atoms/ComponentWrapper.ts:5]: Why did you do this? There's already a `?`.

File: apps/client/src/components/RegisterLookupSummary.tsx [hunk 2]
```diff
@@ -14,23 +14,34 @@ import {
 import AddressLines from "./AddressLines";
 import ChangeAddressModal from "./Location/ChangeAddressModal";
 
-// @TODO: Convert this file to TS
-// @TODO replace unused props from RegisterLookupSummary in LocationResult
-const RegisterLookupSummary = ({
+// @TODO: can you make a props file for these kinds of props, @andre?
```
Comment (afjlambert) [inline apps/client/src/components/RegisterLookupSummary.tsx:17]: Good question. Can you remove it from the source code however? I'll add it to the list of stuff to improve.

File: apps/client/src/components/RegisterLookupSummary.tsx [hunk 3]
```diff
@@ -14,23 +14,34 @@ import {
...
+type zoningPlanProps = {
+  name: string;
+};
+
+type RegisterLookupSummaryProps = {
+  addressFromLocation: any;
+  compact: boolean;
+  setActiveState: Function;
+  topic: any; // @TODO: replace with custom hooks
```
Comment (afjlambert) [inline apps/client/src/components/RegisterLookupSummary.tsx:26]: Please improve comment...
`// @TODO: Replace it with IMTR-Client's TopicType`
?

File: apps/client/src/components/RegisterLookupSummary.tsx [hunk 4]
```diff
@@ -41,7 +52,7 @@ const RegisterLookupSummary = ({
   }
 
   return (
-    <ComponentWrapper marginBottom={hasIMTR ? "0" : null}>
+    <ComponentWrapper marginBottom={hasIMTR && 0}>
```
Comment (afjlambert) [inline apps/client/src/components/RegisterLookupSummary.tsx:55]: 👍 nice

File: apps/client/src/components/RegisterLookupSummary.tsx [hunk 5]
```diff
@@ -14,23 +14,34 @@ import {
...
   topic: { slug, hasIMTR },
 }) => {
-  const sessionContext = useContext(SessionContext);
+  // @TODO: replace with custom topic hooks
+  const sessionContext = useContext<SessionDataType>(SessionContext);
   const address = addressFromLocation
     ? addressFromLocation
-    : sessionContext[slug].address;
+    : sessionContext[slug].address; // @TODO: replace with custom hooks
   const { restrictions, zoningPlans } = address;
```
Comment (afjlambert) [inline apps/client/src/components/RegisterLookupSummary.tsx:40]: I believe if we type `Session` then address and zoningPlan's will all inherit their types.
Can you add a comment?
`// @TODO; type Session and/or Address and remove types of zoningPlan below`

**Review by afjlambert — CHANGES_REQUESTED** [review]

**Pull request merged**